source/extensions/filters/network/kafka/generated/kafka_request_resolver.cc

```diff
@@ -0,0 +1,36 @@
+// DO NOT EDIT - THIS FILE WAS GENERATED
```

Comment:
@mattklein123 Actual decision is yours, as then you are requiring to have jvm & gradle installed (as you need to get Kafka's Java library to get the protocol data, so you need Java - there is *no* BNF with Kafka spec, the BNF was created from Java code).

Erlang https://github.com/klarna/kafka_protocol#code-generation chose the other way - the generated files are committed.

It's up to you if you want to pull java&related tooling into your build ecosystem - I'm fine either way.

Reply:
We use Bazel. It's there. Yes I would prefer to make it hermetic. I think this can likely be done as an external genrule Bazel target.

Reply:
You may be interested in https://github.com/apache/kafka/pull/5893
Where we added a real protocol-generator to Apache Kafka. It will take us a while to get all our request/response pairs, but it will be great if we'll generate from same definitions.

Reply:
so with request/resp spec in json, we can remove java from ecosystem
I'll make rewrite the existing generator to python and invoke it as part of build

Reply:
@gwenshap FYI - if this gets merged in, we will depend on the fact that spec files present at https://github.com/apache/kafka/tree/2.2/clients/src/main/resources/common/message are named `*Request.json` (and in future `*Response.json` too)